api/external.go

```diff
@@ -32,7 +33,7 @@ func (a *API) ExternalProviderRedirect(w http.ResponseWriter, r *http.Request) e
 	providerType := r.URL.Query().Get("provider")
 	provider, err := a.Provider(ctx, providerType)
 	if err != nil {
-		return badRequestError("Unsupported provider: %+v", err)
+		return badRequestError("Unsupported provider: %+v", err).WithInternalError(err)
```

Comment:
shouldn't these be ` badRequestError("Unsupported provider: %s", providerType).WithInternalError(err)` not the error being passed to the client?

Reply:
Yeah, it seems so.